darwin/client.py

```diff
@@ -17,21 +20,43 @@
     Unauthorized,
 )
 from darwin.utils import is_project_dir, urljoin
-from darwin.validators import name_taken, validation_error
 
 
 class Client:
-    def __init__(self, config: Config, default_team: Optional[str] = None):
+    def __init__(self, config: Config, log: Logger, default_team: Optional[str] = None):
         self.config: Config = config
         self.url: str = config.get("global/api_endpoint")
         self.base_url: str = config.get("global/base_url")
         self.default_team: str = default_team or config.get("global/default_team")
         self.features: Dict[str, List[Feature]] = {}
         self._newer_version: Optional[DarwinVersionNumber] = None
+        self.log = log
 
-    def get(
-        self, endpoint: str, team: Optional[str] = None, retry: bool = False, raw: bool = False, debug: bool = False
-    ) -> Union[Any, requests.Response]:
+    def _get_raw(self, endpoint: str, team: Optional[str] = None, retry: bool = False) -> Response:
+        response: Response = requests.get(urljoin(self.url, endpoint), headers=self._get_headers(team))
+
+        self.log.debug(
+            f"Client get request response ({response.json()}) with unexpected status "
+            f"({response.status_code}). "
+            f"Client: ({self})"
+            f"Request: (endpoint={endpoint})"
+        )
+
+        if response.status_code == 401:
+            raise Unauthorized()
+
+        if response.status_code == 404:
+            raise NotFound(urljoin(self.url, endpoint))
+
+        if response.status_code != 200 and retry:
```

Comment:
```suggestion
        if not response.ok and retry:
```
^--- I think this is what we want, since 201 is sometimes returned as well